Maple2.Server.Web/Controllers/BlueprintController.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.IO;
+using Maple2.Tools;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Maple2.Server.Web.Controllers;
+
+[Route("/blueprint/ms2/01/")]
+public class BlueprintController : ControllerBase {
+
+    [HttpGet("{blueprintId}/{ugcUid}.png")]
+    public IResult GetBlueprint(long blueprintId, string ugcUid) {
+        string fullPath = Path.Combine(Paths.WEB_DATA_DIR, "blueprint", blueprintId.ToString(), $"{ugcUid}.png");
```

Comment:
**🚨 suggestion (security):** Consider improving file path security to prevent potential path traversal vulnerabilities.

To enhance security, use Path.GetFullPath() in combination with Path.Combine(), and consider sanitizing the ugcUid parameter. Here's an example:

string safePath = Path.GetFullPath(Path.Combine(Paths.WEB_DATA_DIR, "blueprint", blueprintId.ToString()));
if (!safePath.StartsWith(Paths.WEB_DATA_DIR)) return Results.BadRequest();
string fileName = Path.GetFileName(ugcUid) + ".png";
string fullPath = Path.Combine(safePath, fileName);